src/ros_filter.cpp

```diff
@@ -430,6 +448,119 @@ namespace RobotLocalization
+        }
+        else if(filteredPosition.header.frame_id == mapFrameId_)
+        {
+          try
+          {
+            tf2::Transform worldBaseLinkTrans;
+            tf2::fromMsg(worldBaseLinkTransMsg_.transform, worldBaseLinkTrans);
+
+            tf2::fromMsg(tfBuffer_.lookupTransform(baseLinkFrameId_, odomFrameId_, ros::Time(0)).transform,
+                         odomBaseLinkTrans_);
+
+            /*
+             * First, see these two references:
+             * http://wiki.ros.org/tf/Overview/Using%20Published%20Transforms#lookupTransform
+             * http://wiki.ros.org/geometry/CoordinateFrameConventions#Transform_Direction
+             * We have a transform from mapFrameId_->baseLinkFrameId_, but it would actually transform
+             * a given pose from baseLinkFrameId_->mapFrameId_. We then used lookupTransform, whose
+             * first two arguments are target frame and source frame, to get a transform from
+             * baseLinkFrameId_->odomFrameId_. However, this transform would actually transform data
+             * from odomFrameId_->baseLinkFrameId_. Now imagine that we have a position in the
+             * mapFrameId_ frame. First, we multiply it by the inverse of the
+             * mapFrameId_->baseLinkFrameId, which will transform that data from mapFrameId_ to
+             * baseLinkFrameId_. Now we want to go from baseLinkFrameId_->odomFrameId_, but the
+             * transform we have takes data from odomFrameId_->baseLinkFrameId_, so we need its
+             * inverse as well. We have now transformed our data from mapFrameId_ to odomFrameId_.
+             * However, if we want other users to be able to do the same, we need to broadcast
+             * the inverse of that entire transform.
+            */
+
+            mapOdomTrans_.mult(worldBaseLinkTrans, odomBaseLinkTrans_);
+
+            mapOdomTransMsg_.transform = tf2::toMsg(mapOdomTrans_);
+            mapOdomTransMsg_.header.stamp = filteredPosition.header.stamp + tfTimeOffset_;
+            mapOdomTransMsg_.header.frame_id = mapFrameId_;
+            mapOdomTransMsg_.child_frame_id = odomFrameId_;
+
+            worldTransformBroadcaster_.sendTransform(mapOdomTransMsg_);
+          }
+          catch(...)
+          {
+            ROS_ERROR_STREAM("Could not obtain transform from " << odomFrameId_ << "->" << baseLinkFrameId_);
+          }
+        }
+        else
+        {
+          ROS_ERROR_STREAM("Odometry message frame_id was " << filteredPosition.header.frame_id <<
+                           ", expected " << mapFrameId_ << " or " << odomFrameId_);
+        }
+
+        // Fire off the position and the transform
+        positionPub_.publish(filteredPosition);
+      }
+  }
+
+  template<typename T>
+  void RosFilter<T>::integrationLoop()
+  {
+    // Not quite sure how to set the limits on the frequency in this case
+    // Set up the frequency diagnostic
+    double minFrequency = frequency_ - 50;
```

Comment:
Perhaps this 50 should be parameterized? I'll have more to say about this in a general comment.